### ISSUE TITLE
Single frame flash of top of page on scroll start (Recurrence of #413)

### ISSUE THREAD

**Issue by rdarcy1** [cferdinandi/smooth-scroll#454]
#413 seems to have reappeared in v15, fixes the issue if I downgrade to v14.

Tested versions: 15.0.0 (current latest on NPM), 15.1.2 (via CDN)

Test case:
```js
const recordFrames = (callback, autoStart = true) => {
    let running = true;
    let raf;
    const stop = () => {
        running = false;
        cancelAnimationFrame(raf)
    };
    const start = () => {
        running = true;
        run()
    };
    const run = () => {
        raf = requestAnimationFrame(() => {
            callback();
            if (running) run()
        })
    };
    if (autoStart) start();
    return { start, stop }
};

const rf = recordFrames(() => {
    console.log(window.scrollY)
}, false);

rf.start();


new SmoothScroll('a[href*="#"]', {
    speed: 500,
});
```

Output log shows jumps to Y position 0 before scrolling down the page.

Tested on browsers:
Firefox 63
Safari 12
Chrome 70

**Comment by cferdinandi** [comment]
I'm so sorry, but I'm unable to replicate. Tested in Chrome and Firefox. Here's a reduced test: https://codepen.io/cferdinandi/pen/XorNQr.

Can you provide one showing the issue?

**Comment by rdarcy1** [comment]
Sure, this pen replicates the issue for me: https://codepen.io/RobinKlever/pen/OrLWBd

I think it's related to the fixed positioning of the links – the issue doesn't occur if I make the links relative and put them between each section.

**Comment by cferdinandi** [comment]
Forgive me but... I'm still not seeing it? I added your logging to that demo here: https://codepen.io/cferdinandi/pen/vvBxMB

**Comment by rdarcy1** [comment]
Hmm maybe platform specific? Have you tried clicking a link repeatedly? Seems to occur when it's trying to navigate to the position that it's already in (it was happening on every click in the site I'm developing, otherwise would have mentioned this sooner!).

I'm still seeing the issue on your latest pen, as is my colleague. Attached screenshot is of the logs when clicking the section 2 link repeatedly – it's jumping to the top for a frame and then back down.

For info I'm on MacOS 10.14.1 (Mojave), browsers as in my original post.

<img width="488" alt="screenshot 2018-12-07 at 17 05 52" src="https://user-images.githubusercontent.com/15962421/49661942-71460580-fa42-11e8-9c5d-0c492363f49e.png">

**Comment by cferdinandi** [comment]
OHHHHH... so the problem happens if you click the link again (or a different one) before the animation has completed?

**Comment by cferdinandi** [comment]
THAT makes sense, and now I can start to isolate what's going on here. Sorry, that one detail was lost on me earlier and helps me dig in and find the issue.

**Comment by rdarcy1** [comment]
Sorry for not being more specific earlier, only just worked out that it's repeated clicking that's the cause. I had some other scrolling JS in my dev site that I think was making it jump on every click.

I don't think it's necessarily if you click before the animation has completed – I can click section 2, wait 60 seconds and click again and it still jumps. Seems to be if it's trying to navigate to the anchor tag that it's already scrolled to.

**Comment by cferdinandi** [comment]
Alright, this has been fixed with v15.1.3: https://github.com/cferdinandi/smooth-scroll/releases/tag/v15.1.3
